wear/src/main/java/io/homeassistant/companion/android/sensors/HealthServicesSensorManager.kt

```diff
@@ -90,21 +139,61 @@ class HealthServicesSensorManager : SensorManager {
-                    when (info.userActivityState) {
-                        UserActivityState.USER_ACTIVITY_EXERCISE -> "mdi:run"
-                        UserActivityState.USER_ACTIVITY_ASLEEP -> "mdi:sleep"
-                        UserActivityState.USER_ACTIVITY_PASSIVE -> "mdi:human-handsdown"
-                        else -> userActivityState.statelessIcon
-                    },
+                    getActivityIcon(info),
                     mapOf(
                         "time" to info.stateChangeTime,
                         "exercise_type" to info.exerciseInfo?.exerciseType?.name
-                    )
+                    ),
+                    forceUpdate = info.userActivityState == UserActivityState.USER_ACTIVITY_EXERCISE
                 )
 
                 SensorWorker.start(latestContext)
             }
 
+            override fun onNewDataPointsReceived(dataPoints: DataPointContainer) {
+                Log.d(TAG, "New data point received: ${dataPoints.dataTypes}")
+                val floorsDaily = dataPoints.getData(DataType.FLOORS_DAILY)
+                val bootInstant =
+                    Instant.ofEpochMilli(System.currentTimeMillis() - SystemClock.elapsedRealtime())
+                var latest = 0
+                var lastIndex = 0
+
+                dataPoints.dataTypes.forEachIndexed { _, dataType ->
+
+                    if (dataType is DeltaDataType<*, *>) {
+                        val data = dataPoints.getData(dataType)
+                        data.forEachIndexed { indexPoint, dataPoint ->
+                            if (dataPoint is IntervalDataPoint) {
+                                val endTime = dataPoint.getEndInstant(bootInstant)
+                                Log.d(
+                                    TAG,
+                                    "Data for ${dataType.name} index: $indexPoint with value: ${dataPoint.value} end time: ${endTime.epochSecond}"
+                                )
+                            }
+                        }
+                    }
+                }
+                if (floorsDaily.isNotEmpty()) {
+                    floorsDaily.forEachIndexed { index, intervalDataPoint ->
+                        val endTime = intervalDataPoint.getEndInstant(bootInstant)
+                        Log.d(TAG, "Daily Floors data index: $index with value: ${intervalDataPoint.value} end time: ${endTime.epochSecond}")
+                        if (endTime.epochSecond > latest) {
+                            latest = endTime.epochSecond.toInt()
+                            lastIndex = index
+                        }
+                    }
+                    onSensorUpdated(
+                        latestContext,
+                        dailyFloors,
+                        floorsDaily[lastIndex].value,
+                        dailyFloors.statelessIcon,
+                        mapOf()
+                    )
+
+                    SensorWorker.start(latestContext)

```

Comment:
For when the app might collect more data ponts, I think this should be moved out of the daily floors if branch.

Reply:
Well I want to make sure we have the data point because there have been times where we get no data even though the method was called.

Reply:
I see, in that case this is a good solution for now.